nx/lib/nx.ex

```diff
@@ -6408,15 +6408,160 @@ defmodule Nx do
 
   """
   @doc type: :ndim
-  def dot(t1, axes1, t2, axes2) do
+  def dot(t1, axes1, t2, axes2), do: dot(t1, axes1, [], t2, axes2, [])
+
+  @doc """
+  Computes the dot-product of two batched tensors, throughout the batch dimensions.
```

Comment:
I would do something like: "Computes a general dot product with batch axes and contracting axes, where both tensors are vectorized along the given batch axes and contracted along the given contracting axes"